File: v131016/TeeNew/Contour_Levels.pas
unit Contour_Levels;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, TeeSurfa, TeeProcs, Chart;

type
  TContourLevels = class(TBaseForm)
    Series1: TContourSeries;
    CheckBox1: TCheckBox;
    procedure FormCreate(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
  private
    { Private declarations }
    Procedure AddCustomLevels;
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TContourLevels.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(20);
  AddCustomLevels;
end;

procedure TContourLevels.CheckBox1Click(Sender: TObject);
begin
  Series1.AutomaticLevels:=not CheckBox1.Checked;
  if not Series1.AutomaticLevels then AddCustomLevels;
end;

Procedure TContourLevels.AddCustomLevels;
begin
  Series1.NumLevels:=10;
  Series1.CreateAutoLevels;
  With Series1.Levels do
  begin
    Items[0].UpToValue:=-1;
    Items[1].UpToValue:=-0.8;
    Items[2].UpToValue:=-0.6;
    Items[3].UpToValue:=-0.4;
    Items[4].UpToValue:=-0.2;
    Items[5].UpToValue:= 0;
    Items[6].UpToValue:= 0.2;
    Items[7].UpToValue:= 0.4;
    Items[8].UpToValue:= 0.6;
    Items[9].UpToValue:= 0.8;
  end;
end;

initialization
  RegisterClass(TContourLevels);
end.

File: TeeNew/Builder2010/Contour_Levels.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Contour_Levels.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeSurfa"
#pragma resource "*.dfm"
TContourLevelsForm *ContourLevelsForm;
//---------------------------------------------------------------------------
__fastcall TContourLevelsForm::TContourLevelsForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TContourLevelsForm::AddCustomLevels()
{
  Series1->NumLevels = 10;
  Series1->CreateAutoLevels();
  TContourLevels *Levels = Series1->Levels;
  Levels->Items[0]->UpToValue = -1.0;
  Levels->Items[1]->UpToValue = -0.8;
  Levels->Items[2]->UpToValue = -0.6;
  Levels->Items[3]->UpToValue = -0.4;
  Levels->Items[4]->UpToValue = -0.2;
  Levels->Items[5]->UpToValue =  0.0;
  Levels->Items[6]->UpToValue =  0.2;
  Levels->Items[7]->UpToValue =  0.4;
  Levels->Items[8]->UpToValue =  0.6;
  Levels->Items[9]->UpToValue =  0.8;
}
//---------------------------------------------------------------------------
void __fastcall TContourLevelsForm::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(20);
  AddCustomLevels();
}
//---------------------------------------------------------------------------

void __fastcall TContourLevelsForm::CheckBox1Click(TObject *Sender)
{
  Series1->AutomaticLevels = !CheckBox1->Checked;
  if (!Series1->AutomaticLevels) AddCustomLevels();
}
//---------------------------------------------------------------------------

